src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java

```diff
@@ -0,0 +1,91 @@
+import frc.trigon.robot.misc.ShootingCalculations;
+import org.littletonrobotics.junction.Logger;
+import org.trigon.utilities.mirrorable.MirrorableRotation2d;
+
+/**
+ * A command to visualize note shooting.
+ * This command will get the physical information from subsystems when we begin the shot, and calculate the note's position at each timestamp using physics.
+ */
+public class VisualizeNoteShootingCommand extends Command {
+    private static final ShootingCalculations SHOOTING_CALCULATIONS = ShootingCalculations.getInstance();
+    private double startingTimeSeconds;
+    private Translation3d fieldRelativeNoteExitPointTranslation;
+    private Rotation2d startingPitch;
+    private Translation2d initialXYVelocity;
+    private double initialZVelocity;
+    private double noteZ;
+
+    @Override
+    public void initialize() {
+        configureStartingStats();
+    }
+
+    @Override
+    public void execute() {
+        final double timeDifference = Timer.getFPGATimestamp() - startingTimeSeconds;
+
+        final Transform3d noteTransform = calculateNoteTransform(timeDifference);
+        final Pose3d notePose = new Pose3d(fieldRelativeNoteExitPointTranslation, new Rotation3d()).plus(noteTransform);
+        noteZ = notePose.getTranslation().getZ();
+        Logger.recordOutput("Poses/GamePieces/ShotNotePose", notePose);
+    }
+
+    @Override
+    public boolean isFinished() {
+        return noteZ < 0;
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        Logger.recordOutput("Poses/GamePieces/ShotNotePose", new Pose3d(0, 0, 50, new Rotation3d()));
+    }
+
+    private Translation2d calculateInitialXYVelocityWithRobotVelocity(Rotation2d startingPitch, double startingTangentialVelocity, Rotation2d currentRobotAngle) {
+        final Translation2d robotFieldRelativeVelocity = SHOOTING_CALCULATIONS.getRobotFieldRelativeVelocity().toTranslation2d();
+        final double noteXYVelocityRelativeToRobot = startingPitch.getCos() * startingTangentialVelocity;
+        return robotFieldRelativeVelocity.plus(new Translation2d(noteXYVelocityRelativeToRobot, currentRobotAngle.minus(new Rotation2d(Math.PI))));
+    }
+
+    private Transform3d calculateNoteTransform(double timeDifference) {
+        return new Transform3d(calculateNoteXDifference(timeDifference), calculateNoteYDifference(timeDifference), calculateNoteZDifference(timeDifference), new Rotation3d(0, -startingPitch.getRadians(), 0));
+    }
+
+    private double calculateNoteXDifference(double t) {
+        return initialXYVelocity.getX() * t;
+    }
+
+    private double calculateNoteYDifference(double t) {
+        return initialXYVelocity.getY() * t;
+    }
+
+    private double calculateNoteZDifference(double t) {
+        return (initialZVelocity * t) + (-0.5 * ShootingConstants.G_FORCE * t * t);
+    }
+
+    private Pose3d calculateFieldRelativeNotExitPointPose(Pose2d currentRobotPose, Rotation2d currentRobotAngle) {
```

Comment:
Make this return a translation2d, you don't need it to be a pose. 
Seperate into more variables inside this function. 
This should be below the function `configureStartingStats()` since it uses it.